fncas/optimize.h

```diff
@@ -187,7 +187,7 @@ OptimizationResult ConjugateGradientOptimizer<F>::Optimize(const std::vector<dou
   std::vector<double> current_point(starting_point);
 
   const auto initial_f_gradf = gi(current_point);
-  std::vector<double> current_grad = initial_f_gradf.gradient;
+  std::vector<double> current_grad = initial_f_gradf;
```

Comment:
Now it's not `f_gradf` but just a gradient, so I'd suggest `s/initial_f_gradf/initial_grad/`